pennylane/devices/qutrit_mixed/sampling.py

```diff
@@ -0,0 +1,368 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Code relevant for sampling a qutrit mixed state.
+"""
+from typing import List, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane import math
+from pennylane.measurements import (
+    Shots,
+    SampleMeasurement,
+    SampleMP,
+    CountsMP,
+)
+from pennylane.typing import TensorLike
+
+from .utils import QUDIT_DIM, get_num_wires
+from .measure import measure
+from .apply_operation import apply_operation
+
+
+def _group_measurements(mps: List[Union[SampleMeasurement]]):
```

Comment:
For `default.qubit`, we had a constraint that commuting observables should be pulled from the same raw samples.  I'm not sure if this is particularly going to a product constraint for `qutrit.mixed`.

While this is now written, it might still be worth considering whether or not having the infrastructure and boilerplate for a feature we aren't needing.

It might be worth looking at the state of the corresponding `sampling.py` file before we added in this grouping patch:

https://github.com/PennyLaneAI/pennylane/blob/cbcb376bc4004023e4c2e81402c025d75ef14132/pennylane/devices/qubit/sampling.py

Reply:
I see, I can do that if you think that would be better, the only concern I have is, is it worth retaking the samples for each MP with no observable?

Reply:
@albi3ro, @glassnotes I removed the grouping